DEVDOCS-3567: [update] Add category examples to Stencil frontmatter's GraphQL reference (#1503)
bigcommerce/dev-docs: docs/stencil-docs/reference-docs/front-matter-reference.md

|`product_results`|`limit` defines the number of product search results displayed per page. The range of possible values is 1–100 products.|
  
## GraphQL attributes

> You can add [GraphQL Storefront API](/api-docs/storefront/graphql/graphql-storefront-api-overview) queries to your theme via the front matter block in a template file. For example, you can request a product's variants by augmenting the existing [product.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/product.html), as well as a category's products by augmenting the existing [category.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/category.html): 

Reviewer: ```suggestion
You can add [GraphQL Storefront API](/api-docs/storefront/graphql/graphql-storefront-api-overview) queries to your theme using the front matter block in a template file. For example, you can request a product's variants by augmenting the existing [product.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/product.html), or a category's products by augmenting the existing [category.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/category.html): 

```

remove via, change `as well as` to `or` to further support the narrative that these are examples & there are a lot more possibilities.

  }
}"
```

<!--
type: tab
title: Category
-->

 ```yaml title="Front matter block in category.html template" lineNumbers
category:
    shop_by_price: true
    products:
        limit: {{theme_settings.categorypage_products_per_page}}

> gql: "query categoryById ($categoryId: Int!){

Reviewer: yeah, testing is a good idea. are we sure that the `gql` attribute is all the way left?

	site {
    category(entityId: $categoryId) {
      products {
        edges {
          node {
            sku
          }
        }
      }
    }
  }
}"
```

<!-- type: tab-end -->

> We suggest testing GraphQL queries using the [storefront API playground](https://developer.bigcommerce.com/graphql-playground) to refine them before adding them to your template. You can launch the playground in the context of your store by clicking the **Storefront API Playground** link under the **Advanced Settings** menu in your store's control panel.

Reviewer: ```suggestion
We suggest testing and refining GraphQL queries before adding them to your template. You can use the [GraphQL Storefront API playground](https://developer.bigcommerce.com/graphql-playground). To launch the playground in the context of your store, navigate to the **Advanced Settings** menu in your store control panel and click the **Storefront API Playground** link.

```

@markcmurphy and I decided Friday that it's the GraphQL Storefront API, so made it say that. Also revised to leave more space for devs who have their own testing environments to drop out after we say "test it fam".

> Once you have added a query to your template's front matter block, execution happens automatically when the page loads. The data returned by the query will be returned in the page's context and made available to the handlebars under the `gql` key. The following examples illustrate how you can retrieve the variant data from the above query in `product.html`, as well as a category's product data from the above query in `category.html`:

Reviewer: we can do less in this paragraph.

Handlebars is the proper name of a library, so it should always be capitalized. `the handlebars` should be Handlebars; when `Handlebars` appears following a definite article, Handlebars is an adjective modifying `object`, `implementation`, `context`, or similar. https://handlebarsjs.com/

```suggestion
Templates with queries defined in their front matter automatically execute their queries on page load. You can access the returned data using the `gql` Handlebars object in the page's context. The following examples illustrate how to access `gql` object data using results from the previously-defined front matter queries:

```

<!--
type: tab
title: Product
-->

> ```handlebars title="Handlebar in product.html template" lineNumbers

Reviewer: ```suggestion
```handlebars title="Example gql Handlebars object in product.html template" lineNumbers

```

{{/each}}
{{/if}}
```

<!--
type: tab
title: Category
-->

> ```handlebars title="Handlebar in category.html template" lineNumbers

Reviewer: ```suggestion
```handlebars title="Example gql Handlebars object in category.html template" lineNumbers

```
